### PR TITLE
fix #1024: add server option: 'ignoreUncaughtExceptions'

### LINKED ISSUE #1024: Add option to disable restify's uncaughtException handler
I would like to be able to disable (turn off) restify's handling for uncaught exceptions.

This is so the node server when used with --abort-on-uncaught-exception will core dump (as currently restify will handle the exception, report and error to that request and then continue on processing).

### PR DESCRIPTION
**Pull request by twhiteman**
Allows the user to disable the uncaughtException error handler by setting *ignoreUncaughtExceptions* when creating the restify server. The default behaviour is unchanged (i.e. uncaught exceptions are still enabled when this option is not supplied).

Note: I added tests for this, but it felt a little awkward as it had to use a separate process to run the restify server - this is because disabling the uncaught exception handler will cause node to abort the process when an uncaught error occurs.

### CODE REVIEW

File: docs/index.restdown [hunk 1]
```diff
@@ -143,6 +143,7 @@ below (and `listen()` takes the same arguments as node's
 ||certificate||String||If you want to create an HTTPS server, pass in the path to PEM-encoded certificate and key||
 ||key||String||If you want to create an HTTPS server, pass in the path to PEM-encoded certificate and key||
 ||formatters||Object||Custom response formatters for `res.send()`||
+||ignoreUncaughtExceptions||Boolean||When true (default is false), turns off restify's uncaughtException handler||
```
Comment (yunong) [inline docs/index.restdown]: Can we reverse this so the default is to turn it off? I'd rather restify ship out of the box to _not_ swallow errors.

**Comment by yunong** [comment]
@twhiteman thanks so much for doing this. Everything looks great -- just added a comment about reversing the defaults. I think we want to ship this so that by default we _don't_ catch exceptions.

**Comment by twhiteman** [comment]
@yunong Whilst I myself would prefer this to default to on (for Joyent's usage), I think it would be unexpected for the bulk of Restify's users, as they'd be wondering why their server is now crashing (instead of responding with an error response) for any unexpected Restify errors.

**Comment by yunong** [comment]
I think we would gate this behind a major version change -- and update the change notes.

**Comment by twhiteman** [comment]
I could go for that (i.e. with a major version change and docs) - certainly makes sense with node domain already being deprecated.

Anyone else have thoughts for this? If not, I'll update the PR.

**Comment by micahr** [comment]
I'd second the desire to have the default be not to catch exceptions.

**Comment by DonutEspresso** [comment]
+1 this looks great. Thanks @twhiteman!

**Comment by yunong** [comment]
@micahr I think this needs to merge into 5.x?

**Comment by micahr** [comment]
Yes, 5.x would be the best place to put it.

**Comment by yunong** [comment]
@twhiteman would you mind changing this to pull into the 5.x branch?

**Comment by twhiteman** [comment]
Okay - I've created a new PR (#1034) to target the 5.x branch and enable this feature by default. Closing this one.

**Pull request closed without merging**